src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java

```diff
@@ -79,23 +74,9 @@
  * An imported type has the same name as a declaration, such as a member variable.
  * </li>
  * <li>
- * There are two or more imports with the same name.
+ * There are two or more static imports with the same method name.
  * </li>
  * </ul>
- * <p>
- * For example, in the following case all imports will not be flagged as unused:
- * </p>
- * <pre>
- * import java.awt.Component;
- * import static AstTreeStringPrinter.printFileAst;
- * import static DetailNodeTreeStringPrinter.printFileAst;
- * class FooBar {
- *   private Object Component; // a bad practice in my opinion
- *   void method() {
- *       printFileAst(file); // two imports with the same name
```

Comment:
We need to put this line in all examples.
Or .. are we sure that Check skip static imports of same name? So real usage is not required.

Can you prove that this works like this by CLI?

Reply:
If there are two static imports of same name and if the method is called, compiler itself gives error. 
`error: reference to method is ambiguous`

But if there are two static imports with same name but the method is not called then it will be a violation for not being used, so can't verify this behavior.

Reply:
Please share CLI example of javac that gives such error.

Reply:
```
/var/tmp $ javac MyClass.java
6: error: reference to method is ambiguous
		method();
		^
  both method method() in Demo and method method() in Test match
1 error
[Finished in 947ms]

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name = "TreeWalker">
    	<module name="UnusedImports"/>
    </module>
</module>

/var/tmp $ cat MyClass.java
import static demo.Test.method;
import static demo.Demo.method; 

class MyClass{
	public static void main(String args[]){
		method();
	}
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml YOUR_FILE.java
Starting audit...
Audit done.
```

Reply:
javac is giving an compilation error but Check is passing for these imports.

Reply:
they are not static.

@TanayMorakhia , please use this two classes they are from jdk (so javac will be good):
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Calendar.html#getInstance()
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/text/DateFormat.html#getInstance()

Reply:
okay

Reply:
> they are not static

@romani is this in response to my comment at https://github.com/checkstyle/checkstyle/pull/13020#discussion_r1184777111? If so, please explain.

Reply:
I am not sure how I searched last time, my mistake. They are static too. Ok to use.

https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Map.html#of()
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/List.html#of()

Reply:
@TanayMorakhia, please let us know if you are confused on what  we asking.

Reply:
I am a little busy with my college work, will start working on this in a day or two.

Reply:
please share java file your used for JDK compiler.

we should not use any non compileable file in our examples.

Reply:
@stoyanK7 , you added this at https://github.com/checkstyle/checkstyle/commit/ad27aa0feb4804cf3a14430b2cb90c3d13dbe532
are we confident that this is compile-able code ?

Reply:
`import java.util.Calendar;
import java.text.DateFormat;
import static java.util.Calendar.getInstance;
import static java.text.DateFormat.getInstance;

class MyClass{
	public static void main(String args[]){
		Calendar c = getInstance();
		DateFormat e = getInstance();
	}
}`

Also tried to compile with Map#of and List#of function, it also gave the same error.

for reference : JDK version - 17.0.7

Reply:
@TanayMorakhia @romani 
The code at #12224 is compileable. The catch is that the `emptyIfNull()` methods have slightly different signatures.
- [SetUtils.emptyIfNull(Set\<T> set)](https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/SetUtils.html)
- [CollectionUtils.emptyIfNull(Collection\<T> collection)](https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/CollectionUtils.html)

Here is a local run:
```
                                                                            
$ cat src/main/java/org/example/Test1.java 
package org.example;

import java.util.Set;

public class Test1 {
    public static void sameName(Set<Integer> set) {
        System.out.println("Test1 sameName()");
    }
}
                                                                             
$ cat src/main/java/org/example/Test2.java 
package org.example;

import java.util.Collection;

public class Test2 {
    public static void sameName(Collection<Integer> collection) {
        System.out.println("Test2 sameName()");
    }
}
                                                                               
$ cat src/main/java/org/example/Main.java 
package org.example;

import java.util.Set;

import static org.example.Test1.sameName;
import static org.example.Test2.sameName;


public class Main {
    public static void main(String[] args) {
        System.out.println("Hello world!");
        sameName(Set.of(1, 2));
    }
}   
                                                                                                                                                                                                                           
$ mvn compile                             
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------< org.example:test-static-imports-11 >-----------------
[INFO] Building test-static-imports-11 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ test-static-imports-11 ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 0 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ test-static-imports-11 ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to /home/stoyan/temp/checkstyle/test-static-imports-11/target/classes
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.023 s
[INFO] Finished at: 2023-05-12T12:17:54+02:00
[INFO] ------------------------------------------------------------------------
                                                                           
$ mvn exec:java -Dexec.mainClass=org.example.Main
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------< org.example:test-static-imports-11 >-----------------
[INFO] Building test-static-imports-11 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:3.1.0:java (default-cli) @ test-static-imports-11 ---
Hello world!
Test1 sameName()
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.376 s
[INFO] Finished at: 2023-05-12T12:17:59+02:00
[INFO] ------------------------------------------------------------------------

```

Reply:
Can't find any such methods, most methods with same name are for data structures which are from the same interface so have the same signature.
Should I make the suggested change? it looks correct to me.

Reply:
Please do

Reply:
@stoyanK7 , please take a look. Maybe you will find. It will be a problem for us later on in summer when we try to  make all tests compilable.

Reply:
@TanayMorakhia 

Do those 2 work?

Arrays.copyOf - https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Arrays.html
Map.copyOf - https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html

Reply:
yes! these works as it has different signatures.

Reply:
@TanayMorakhia , please push changes.